### PR TITLE
GROUP <property> BY <function>

### PR DESCRIPTION
**Pull request by taegyunkim**
Modified grammar/codegen to support `GROUP <property> BY <user function>`

### CODE REVIEW

File: src/code_gen.rs [hunk 1]
```diff
@@ -235,64 +232,40 @@ impl<'a> TreeFold<'a> for CodeGen<'a> {
...
-                            .iter()
-                            .map(|s| String::from("\"") + s + "\"")
-                            .collect::<Vec<String>>()
-                            .join(", "),
-                    );
-                    parts.push_str("}");
-
-                    let block = format!(
-                        "node_ptr = get_node_with_id(target, mapping->at(\"{node_id}\"));
+                let attribute = &self.config.attributes_to_property_parts[p.id_name];
```
Comment (anirudhSK) [inline src/code_gen.rs:235]: A few comments explaining what is going on here would be useful, both at this location and the locations below.

File: example_queries/group_by/count.cc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// udf_type: Aggregation
+// id: count
+// return_type: int
+
+class count {
+public:
+  int operator()(std::string height) {
```
Comment (anirudhSK) [inline example_queries/group_by/count.cc:7]: Why is height a string here, instead of an integer?

File: example_queries/group_by/count.cc [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// udf_type: Aggregation
```
Comment (anirudhSK) [inline example_queries/group_by/count.cc:1]: histogram.cc might be a more appropriate name for this file.

File: example_queries/group_by/count.cc (hunk 2 above)
Comment (taegyunkim) [reply example_queries/group_by/count.cc:7]: Property is stored using string type. I chose to do so for the simplicity of coding. 
https://github.com/anirudhSK/dyn_tracing/blob/43af6932295b411b7436409e826f914612a4cee1/wasm/graph_utils.h#L17

We can convert it back to integer type or modify property map to store typed values.

File: example_queries/group_by/count.cc (hunk 3 above)
Comment (taegyunkim) [reply example_queries/group_by/count.cc:1]: done.

File: example_queries/group_by/count.cc (hunk 2 above)
Comment (anirudhSK) [reply example_queries/group_by/count.cc]: OK. Let's leave it this way for now and leave a TODO to use typed values subsequently.

Comment (taegyunkim) [reply example_queries/group_by/count.cc]: Ended up adding type conversion statement.

File: src/code_gen.rs (hunk 1 above)
Comment (taegyunkim) [reply src/code_gen.rs:237]: It's just stored in memory, not sent to backend.

Comment (taegyunkim) [reply src/code_gen.rs:237]: Implemented moving average, and added comments here.

**Pull request merged**